dds/idl/langmap_generator.cpp

```diff
@@ -516,6 +516,24 @@ struct GeneratorBase
     return ss.str();
   }
 
+  static std::string generateEqualCxx11(const std::string&, AST_Decl*, const std::string& name, AST_Type* field_type,
+                                        const std::string&, bool, Intro&,
+                                        const std::string&)
+  {
+    std::stringstream ss;
+
+    AST_Type* actual_field_type = resolveActualType(field_type);
+    const Classification cls = classify(actual_field_type);
+    if (cls & (CL_PRIMITIVE | CL_ENUM | CL_STRING | CL_ARRAY | CL_STRUCTURE | CL_UNION | CL_SEQUENCE | CL_FIXED)) {
+      ss <<
+        "    return this->_" << name << " == rhs._" << name << ";\n";
+    } else {
+      std::cerr << "Unsupported type for union element\n";
```

Comment:
If this is supposed to be a warning, you can use:

https://github.com/DOCGroup/ACE_TAO/blob/ab3b192fcda5715a0c92e5627acdeb4e8f71911c/TAO/TAO_IDL/include/utl_err.h#L448

I think it would be:

```suggestion
      idl_global->err()->misc_warning("Unsupported type for union element", node);
```